### PR TITLE
A few more of new typos picked up by codespell 2.2.5

### PR DESCRIPTION
**Pull request by yarikoptic**
no limits to perfection! ;)

### CODE REVIEW

File: Cython/Includes/cpython/object.pxd [hunk 1]
```diff
@@ -231,7 +231,7 @@ cdef extern from "Python.h":
     # used to determine the result of this function.
 
     # Subclass determination is done in a fairly straightforward way,
-    # but includes a wrinkle that implementors of extensions to the
+    # but includes a wrinkle that implementers of extensions to the
```
Comment (h-vetinari) [inline Cython/Includes/cpython/object.pxd:234]: Both versions (implementer/implementor) are [valid](https://www.thefreedictionary.com/implementor).

Comment (yarikoptic) [reply Cython/Includes/cpython/object.pxd:234]: indeed... and was moved into "rare" dictionary in codespell just 3 days ago: https://github.com/codespell-project/codespell/commit/d46b6872ed7be34496b9cc52f444eb6cefdb7c20 . I will just add explicit skip for it for now... will refresh PR tomorrow

Comment (yarikoptic) [reply Cython/Includes/cpython/object.pxd]: done

File: .codespellrc [hunk 2]
```diff
@@ -5,4 +5,5 @@ skip = .git,*.pdf,*.svg,msvc_strings.pyx,cython.st,test_genericclass_exttype.pyx
 ignore-regex= Tripple|/Fo|ist das|TOi|respect to the includee
 # forin - ? tag used in 3 files. may be could be renamed?
 # delimeters - typo, but used in API so kept as is
-ignore-words-list = asend,forin,nd,mis,te,nin,ue,ccompiler,ket,defin,delimeters,filetests,crasher
+# implementor,implementors - was decided that both forms are ok
```
Comment (h-vetinari) [inline .codespellrc:8]: FWIW, I don't make "decisions" here, I just noted that both are valid English. It's up to the maintainers if they want to also eliminate rare(r) spelling variants.

File: Cython/Compiler/Nodes.py [hunk 3]
```diff
@@ -6518,7 +6518,7 @@ class InPlaceAssignmentNode(AssignmentNode):
     #
     #  This code is a bit tricky because in order to obey Python
     #  semantics the sub-expressions (e.g. indices) of the lhs must
-    #  not be evaluated twice. So we must re-use the values calculated
+    #  not be evaluated twice. So we must reuse the values calculated
```
Comment (da-woods) [inline Cython/Compiler/Nodes.py:6521]: As far as I can tell both forms are right so not sure I want to change this (I kind of prefer the hypenated form, although wouldn't correct the other form if it was already there)

**Comment by scoder** [comment]
Just a side note (don't think it applies to the changes at hand): we try to stick to British English spellings for historical reasons. The original code base was written in New Zealand.

**Comment by yarikoptic** [comment]
eh, there is an open issue https://github.com/codespell-project/codespell/issues/103 ... didn't look in detail yet

**Comment by scoder** [comment]
Thanks

**Pull request merged**